### PR TITLE
Base components for card space user page

### PR DESCRIPTION
**Pull request by Aierie**
Base components for card space user page. These components provide basic UI styling + behaviour, and some conveniences for editing. For examples of how these components are used and some other card space UI behaviour, see https://github.com/cardstack/cardstack/pull/2636, which includes some HAX! (see the description for other caveats).

re: `@sufficientContent` in `CardContainer`, this is here so that cards that use the `CardContainer` can decide on their own whether to display themselves (in view-only mode), and display their placeholder content (in editable mode). I'm not sure whether this is good - having sufficient content to not display a placeholder in editable mode might not necessarily mean having sufficient content to display a card in view-only mode, but it makes it easy to specify things like placeholders for these cards at the moment. It may be good to introduce a separate way to decide whether or not to display anything for empty cards in view-only mode (or simply use conditionally render in the consuming component), and leave `@sufficientContent` to help with deciding whether to render placeholder content.

### CODE REVIEW

**Comment by backspace** [comment]
Sorry for the delay, things have been a lot, but I’ll finally review tomorrow! 😯

File: packages/web-client/tests/integration/components/card-space/user-page/layout-card-container-test.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  '[data-test-card-space-card-edit-modal-title]';
+
+module(
+  'Integration | Component | card-space/user-page/layout-card-container',
+  function (hooks) {
+    setupRenderingTest(hooks);
+
+    hooks.beforeEach(function () {
+      this.setProperties({
+        editable: false,
```
Comment (backspace) [inline packages/web-client/tests/integration/components/card-space/user-page/layout-card-container-test.ts:29]: minuscule importance but this is set to `true` in all but one test, could make that the default 🤓

**Review by backspace — APPROVED** [review]
This seems well-factored and thoroughly exercised with tests 🤩 I tried it out with the HAX branch, looks good! It can be hard to know with this kind of thing whether it’ll be a good abstraction in practice but that branch helps show that it probably is, and we can always adjust if needed.

I wonder if there’s a more clear name that `sufficientContent`… `shouldRender`? 🤔

**Comment by Aierie** [comment]
Closing for now. It looks like these components may not be used for the MVP.

**Pull request closed without merging**